### ISSUE TITLE
September 2022: Meetup in Cologne

### ISSUE THREAD

**Issue by Florob** [Rustaceans/rust-cologne#98]
# ToDo

- [x] Settle for a date and location: **Wednesday 2022-09-07, 19:15 CEST** at C4 (default: first Wednesday each month)
- [ ] Send "Save the Date" info (with link to this ticket)
- [X] Basic Agenda (default: Open Space)
- [x] Final invitation (deadline **2022-08-31**)
  - [x] [meetup]
  - [x] [website]
  - [ ] [twitter]
  - [ ] [community calendar]

# Invitation draft

```markdown
Title: Idiomatic Rust: How clippy helps you become a better coder

Dear Rustaceans,

our next Meetup will be on **Wednesday 2022-09-07, 19:15 CEST**.

> clippy is for people who find a certain emptiness inside when they finally get code through the compiler

When writing Rust did you ever feel like "it works, but there must be a better way"?
`clippy` is a linting tool that helps you find and solve common issues when learning Rust.
Especially when coming from another programming language `clippy` will help make the transition
easier by recognizing anti-patterns and pointing out possible logic errors.

`clippy` can be used in a couple of ways:
- protective: "Attention, this is likely not doing what you had in mind."
- instructive: "Did you know there's a new language feature covering your use case?"
- cautious: "Are you sure, this is safe to do?"
- sadistic: "This will make your rocket launch fail whenever your computer gets too close to a black hole on a Friday"

Kai will give you a live-coding tour through `clippy`, showing how it can help you to find bugs,
learn idiomatic programming patterns, discover helpful functions in the `std`-library,
introduce new language features and how it can ruin your code if you go too far.

Any remaining time will be used as **open space**: Imagine a bazaar, where you can suggest topics and spontaneously join any discussion you find interesting. It all depends on your interests. You can enjoy some drinks, meet nice people and discuss about Rust.

**[You can register here](...)**.

We are looking forward to seeing you. :-)

Yours,
Florian and Kai
―  
The meetup will likely be held in German, we will however reevaluate this at the beginning of the evening and may switch to English if needed.
―
Due to COVID-19 still being around there are some precautions we feel are necessary to make everyone at ease.

We have restricted the maximum number of participants to ensure social distancing is possible.
Accordingly we need to know how many people will attend the meetup.
Please make sure to register and, in case you are no longer able to attend, cancel your registration.

Additionally we generally expect participants to wear masks. During warm weather the venue allows for
decent ventilation. In this case participants may **unanimously** decide masking is not necessary.
```
# Rust News
- [new rust release?](https://github.com/rust-lang/rust/blob/stable/RELEASES.md)

**Comment by kawogi** [comment]
I thought about presenting how `clippy` can help you to write idiomatic code. Maybe by trying to trigger as many lints with a short code as possible. - then going berserk and show when to stop enabling warnings (`restriction`)

**Comment by Florob** [comment]
@kawogi Sounds good. Do you have an estimate for how long that would take? Do we need something else?

**Comment by kawogi** [comment]
Let me see … there are 456 lints … one minute for each … jk ;)

I can make it as long as we like it to be and I can abort any time. I think we'll scratch a lot of different topics and hear many different opinions. Those will likely spark a discussion filling the rest of the evening. There's a risk that this format gets boring over some time.
TL;DR we don't need another topic IMO :)

Title suggestion (if this works): "Idio~~ma~~tic Rust"

Y'all can suggest your favorite lints if you like.

**Comment by YtvwlD** [comment]
just a small nitpick: The

> Due to COVID-19 still being around, bringing a face mask will be necessary. For the same reason we need to know how many people will attend the meetup. Please make sure to register and, in case you are no longer able to attend, cancel your registration. Thank you.

doesn't seem to exactly match the

> COVID-19 safety measures:
> * Masks required
> * Event will be indoors

on the meet-up page (which again does not seem to match what actually applies to the event).

I would appreciate this to be merged and kept up-to-date to aid (possible) participants in their risk assessment. :)

**Comment by kawogi** [comment]
@Florob Can you please proof-read the invitation?

Not sure what to do about the COVID-Measures. Maybe we should point out that masks are required to bring with you, but we'll decide on the spot whether they are mandatory to wear (depending on the number of participants.)

**Comment by Florob** [comment]
@YtvwlD Thank you for the feedback. I don't necessarily agree that the text does not match what we specify on meetup. That is mostly due to "Masks required" being vague though. We definitely have to do better with what actually happens at the meetup. I think we were much more upfront with people about this during the July meetup.
The general default we have in mind is definitely masking. With social distancing and proper ventilation (that we can get while it's still warm) I personally don't think it is strictly required. However,I would never make that call for the group. It needs consensus from everybody present, and it needs to be very clear that people can ask the rest of the group to put on masks without feeling bad about it. I would agree that we did not communicate this (clearly) enough in August, and it would be good to communicate it beforehand.

@kawogi Proofread. I made some minor edits to the announcements and some larger ones to the COVID-19 specific section.
